general/releases.rst

```diff
@@ -0,0 +1,146 @@
+and :ref:`optee_test`.
+
+.. _releases_versioning_schema:
+
+Versioning schema
+*****************
+OP-TEE follows `Semantic Versioning 2.0.0`_. What that means in practice is well
+described at the link just shown.
+
+.. _releases_release_procedure:
+
+Release procedure
+*****************
+There are certain steps that needs to be done when making a release. This
+checklist here serves as guidance to the one in charge of making a new release.
+Roughly start with this 2-3 weeks before the targeted release date.
+
+tl;dr
+=====
+.. list-table:: Short version of the OP-TEE release procedure
+    :widths: 60 300 10
+    :header-rows: 1
+
+    * - When
+        (Tminus)
+      - Action
+      - Example
+
+    * - 3w
+      - Create release pull request
+      - `PR#3099`_
+
+    * - 3w
+      - Inform maintainers about upcoming release
+      -
+
+    * - 1w
+      - Create RC-x tag in optee_* + build.git
+      - git tag -a 3.x.y-rc1 -m "3.x.y-rc1"
+
+    * - 1w
+      - Inform maintainers about RC tag
+      -
+
+    * - 1w
+      - Test platform builds / devices
+      - 
+
+    * - Release day
+      - Update CHANGELOG.md_ 
+      - `changelog example`_
+
+    * - Release day
+      - Collect/merge ``Tested-By`` tags
+      - `commit example`_
+
+    * - Release day
+      - Create release branch in :ref:`manifest`
+      - git checkout -b 3.x.y github/master
+
+    * - Release day
+      - Update manifest XML-files
+      - `3.6.0 stable`_
+
+    * - Release day
+      - Create release tag in optee_* + build.git
+      - git tag -a 3.x.y -m "3.x.y"
+
+    * - Release day
+      - Inform maintainers and stakeholder that release has been completed.
+      -
+
+
+Long version
+============
+
+    1. Create a "release pull request" at GitHub ought to collect ``Tested-By``
+       tags from various maintainers. As an example, see `PR#3099`_.
+
+    2. Send email to all maintainers to let them know about the upcoming
+       release.
+       
+    3. Create a release candidate (RC) tag (annotated tag, i.e., ``git tag -a
+       3.x.y-rc1 -m "3.x.y-rc1"``) in the following gits
+       ``optee_{os,client,test,examples,docs}`` and ``build.git``.
+
+    4. Send a follow up email to all maintainers to let them know that there is
+       a release tag ready to be tested on their devices for the platforms that
+       they are maintaining.
+
+    5. In case major regressions are found, then fix those and create a another
+       RC-tag (i.e., repeat step 3 and 4 until there are no remaining issues
+       left).
+
+    6. On release day: Update CHANGELOG.md_ see this `changelog example`_ to see
+       how that should look like.
+
+    7. Collect all tags (``Tested-By`` etc) from maintainers and use those in
+       the commit message, for an example see this `commit example`_.
+
+    8. Create a new branch in :ref:`manifest` where the name corresponds to the
+       release you are preparing. I.e., ``git checkout -b 3.x.y github/master``.
```

Comment:
"Create a new branch from `master`" maybe?
The commands I'm using:
```
export VER=3.x.y
cd .repo/manifest
git fetch
git checkout -b $VER origin/master
```

Reply:
> "Create a new branch from `master`" maybe?

Yes! Fixed in upcoming patch.

> The commands I'm using:
> 
> ```
> export VER=3.x.y
> cd .repo/manifest
> git fetch
> git checkout -b $VER origin/master
> ```

This works, but I'd probably recommend to work with a standalone clone of manifest for this and not do the work in the "hidden" clone in a repo forest. Having said that, `origin` is the correct remote here and not `github` as I've stated. Will fix in upcoming patch.